### PR TITLE
UF-335: Fix NPE when injected SessionInfo used outside of request, an…

### PR DESCRIPTION
**Pull request by mbarkley**
…d check for non-null filesystems in AbstractIOService.

This is a partial for UF-335. See the [issue](https://issues.jboss.org/browse/UF-335) comments for details.

### CODE REVIEW

File: uberfire-server/src/main/java/org/uberfire/server/cdi/UberFireGeneralFactory.java [hunk 1]
```diff
@@ -31,7 +33,11 @@
     @RequestScoped
     @Default
     public static SessionInfo getSessionInfo(AuthenticationService authenticationService) {
-        return new SessionInfoImpl( RpcContext.getQueueSession().getSessionId(), authenticationService.getUser() );
+        return new SessionInfoImpl( getSessionInfo(), authenticationService.getUser() );
+    }
+
+    private static String getSessionInfo() {
+        return (getMessage() != null && getQueueSession() != null ? getQueueSession().getSessionId() : null );
```
Comment (manstis) [inline uberfire-server/src/main/java/org/uberfire/server/cdi/UberFireGeneralFactory.java]: @mbarkley Out of interest what's the significance of ```getMessage()``` it's not used explicitly in ```getSessionInfo()```?

**Comment by manstis** [comment]
+1 to merge.

**Comment by csadilek** [comment]
+1

**Comment by ederign** [comment]
thanks @mbarkley

File: uberfire-server/src/main/java/org/uberfire/server/cdi/UberFireGeneralFactory.java (hunk 1 above)
Comment (mbarkley) [inline uberfire-server/src/main/java/org/uberfire/server/cdi/UberFireGeneralFactory.java]: The queue session is stored in the message returned by `getMessage()`.

Comment (jhrcek) [inline uberfire-server/src/main/java/org/uberfire/server/cdi/UberFireGeneralFactory.java]: So it means there's hidden potential for NPE in every call to `getQueueSession()` when `getMessage()` returns null? Shouldn't that be handled in `getQueueSession()` directly?

**Pull request merged**